File: src/main/javascript/app/view/serviceproblem/WorkItemPanel.js
Ext.define('Spm.view.serviceproblem.WorkItemPanel', {
    extend: 'Ext.panel.Panel',
    alias: 'widget.workItemPanel',

    layout: {
        align: 'stretch',
        padding: 5,
        type: 'vbox'
    },
    title: 'Work Item',

    initComponent: function() {
        var me = this;

        Ext.applyIf(me, {
            items: [
                {
                    xtype: 'container',
                    flex: 0,
                    layout: {
                        align: 'stretch',
                        type: 'hbox'
                    },
                    items: [
                        {
                            xtype: 'fieldcontainer',
                            flex: 1,
                            items: [
                                {
                                    xtype: 'textfield',
                                    fieldLabel: 'Created Date'
                                },
                                {
                                    xtype: 'textfield',
                                    fieldLabel: 'Priority'
                                }
                            ]
                        },
                        {
                            xtype: 'fieldcontainer',
                            flex: 1,
                            items: [
                                {
                                    xtype: 'textfield',
                                    fieldLabel: 'Status'
                                },
                                {
                                    xtype: 'textfield',
                                    fieldLabel: 'Assigned Agent'
                                }
                            ]
                        },
                        {
                            xtype: 'fieldcontainer',
                            flex: 1,
                            items: [
                                {
                                    xtype: 'textfield',
                                    fieldLabel: 'Type'
                                },
                                {
                                    xtype: 'textfield',
                                    fieldLabel: 'Reminder'
                                }
                            ]
                        }
                    ]
                },
                {
                    xtype: 'fieldcontainer',
                    flex: 1,
                    layout: {
                        type: 'fit'
                    },
                    items: [
                        {
                            xtype: 'textfield',
                            fieldLabel: 'Action'
                        }
                    ]
                }
            ]
        });

        me.callParent(arguments);
    }

});

File: bango-test/src/main/java/sonique/bango/driver/panel/serviceproblem/WorkItemPanel.java
package sonique.bango.driver.panel.serviceproblem;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import sky.sns.spm.domain.model.serviceproblem.AssignmentType;
import sky.sns.spm.domain.model.serviceproblem.WorkItemAction;
import sky.sns.spm.domain.model.serviceproblem.WorkItemPriority;
import sky.sns.spm.domain.model.serviceproblem.WorkItemStatus;
import sonique.bango.driver.component.HasTitle;
import sonique.bango.driver.component.form.SupermanPanel;

import java.util.Date;

public class WorkItemPanel extends SupermanPanel implements HasTitle {
    public WorkItemPanel(ServiceProblemTabContent serviceProblemTabContent) {
        super(serviceProblemTabContent, By.cssSelector("[id^='workItemPanel']"));
    }

    public String assignedAgent() {
        return textField("Assigned Agent").value();
    }

    public WorkItemStatus status() {
        return WorkItemStatus.valueOf(textField("Status").value());
    }

    public Date createdDate() {
        return dateField("Created Date", "dd/MM/yyyy HH:mm").value();
    }

    public AssignmentType type() {
        return AssignmentType.valueOf(textField("Type").value());
    }

    public WorkItemPriority priority() {
        return WorkItemPriority.valueOf(textField("Priority").value());
    }

    public WorkItemAction action() {
        return WorkItemAction.getByDescription(textField("Action").value());
    }

    public Date reminder() {
        return dateField("Reminder", "dd/MM/yyyy HH:mm").value();
    }

    public boolean hasNoWorkItem() {
        return contains(By.className("no-work-item-text"));
    }

    @Override
    public String title() {
        WebElement titleElement = element().findElement(By.cssSelector("span.x-header-text"));
        return titleElement.getText();
    }
}
